pubsub.go

```diff
@@ -137,27 +187,43 @@ func (p *PubsubValueStore) Subscribe(key string) error {
 	}
 
 	p.mx.Lock()
-	existingSub, bootstraped := p.subs[key]
+	existingSub, _ := p.subs[key]
 	if existingSub != nil {
 		p.mx.Unlock()
 		sub.Cancel()
 		return nil
 	}
 
 	p.subs[key] = sub
-	ctx, cancel := context.WithCancel(p.ctx)
-	go p.handleSubscription(sub, key, cancel)
+	go p.handleSubscription(sub, key)
 	p.mx.Unlock()
 
+	go p.rebroadcast(key)
 	log.Debugf("PubsubResolve: subscribed to %s", key)
 
-	if !bootstraped {
-		// TODO: Deal with publish then resolve case? Cancel behaviour changes.
-		go bootstrapPubsub(ctx, p.cr, p.host, topic)
-	}
 	return nil
 }
 
+func (p *PubsubValueStore) rebroadcast(key string) {
+	topic := KeyToTopic(key)
+	time.Sleep(p.rebroadcastInitialDelay)
+
+	ticker := time.NewTicker(p.rebroadcastInterval)
+	defer ticker.Stop()
+
+	for {
+		select {
+		case <-ticker.C:
+			val, err := p.getLocal(key)
```

Comment:
What type of bailing and why? It's possible we decided to subscribe to a PubSub topic where no one with a valid record is online and we don't have an old record.

Reply:
Yeah, you're right. I'm just thinking of the case where the record expires, we delete it, and then we keep rebroadcasting forever. It would be nice to cleanup that goroutine but we can deal with that if it becomes an issue.

Reply:
Dealing with it if it becomes an issue seems reasonable. Otherwise, we have to deal with shutting down the goroutine and then spinning a new one up when we get a valid record.

Happy to deal with it if we want, but feels like premature optimization to me, especially since my hope is to increase the rebroadcast duration to be fairly large given that with the `pubsub-get-latest` protocol we barely need it (currently it's only useful for gossipsub)

Reply:
probably the best solution for dealing with this is to have a single goroutine for all the rebroadcasting. This has better scale properties and makes this problem basically disappear.